src/components/grid-aware/StatsBlock/StatsBlock.module.css

```diff
@@ -101,6 +113,12 @@
   line-height: 1.3;
 }
 
+.yoy {
+  color: var(--color-gray-700);
+  font: var(--font-body-small);
+  margin: 5px 0 10px;
```

Comment:
There's a subtle little bug here due to a somewhat obscure rule in CSS regarding margins. CSS has this rule called "collapsing margins" where in certain cases, adjacent margins of adjacent elements may _overlap_ with each other instead of pushing each other apart. Here is a pretty great article with illustrations that explain how margin collapse works: https://www.joshwcomeau.com/css/rules-of-margin-collapse/

For this specific instance, the Figma designs show that when this extra YOY text is present, there should be a 15px gap between the YOY text and the other text below it (the `.statement` element). I think you placed a 10px bottom margin here because there is already a 5px top margin on the other text. However, due to the collapsing margin rule, the 10px margin ends up _overlapping_ with the 5px margin, which means that there ends up only being a 10px space between the text.

There's a few different ways you could avoid this. You can change the bottom margin on the `.yoy` style here to be the full 15px, which will overlap with the other margin and still guarantee that the other text is a full 15px away. If you fix it this way, please add a comment explaining that the margins are being collapsed, since it's difficult to notice without looking at the full HTML + CSS.

Another way to deal with this is to use an [adjacent sibling selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Adjacent_sibling_combinator) to add a custom style to the `.statement` element _only_ when it is preceded by the `.yoy` element. This would look like this:

```css
.yoy + .statement {
  margin-top: 15px;
}
```

The adjacent sibling selector only applies to the latter sibling, so this guarantees that the .statement element has a top margin of 15px _only_ when the .yoy element is just before it.

Reply:
Thanks for helping me understand the margin collapse and adjacent sibling selector! 🤩